src/app/track/track.component.ts

```diff
@@ -1,346 +1,476 @@
-          })
+    createMap() {
+        const tileLayer = new TileLayer({
+            source: new OSM(/*{
+      attributions: [
+        'All maps © <a href="https://www.opencyclemap.org/">OpenCycleMap</a>',
+        ATTRIBUTION
+      ],
+      url: 'https://{a-c}.tile.thunderforest.com/cycle/{z}/{x}/{y}.png?apikey=25763459938a4e9d9bec8bde3481f682',
+    }*/)
+        });
+
+
+        const trackPointsD1: Feature<Geometry>[] = [];
+        const trackPointsD2: Feature<Geometry>[] = [];
+        const trackPointsUntaggedD1: Feature<Geometry>[] = [];
+        const trackPointsUntaggedD2: Feature<Geometry>[] = [];
+        const trackLines: Feature[] = [];
+        const points: Coordinate[] = [];
+
+        if (this.trackData) {
+            let filteredPoints = this.trackData.points.filter(point => point.latitude && point.longitude)
+
+            for (let i = 0; i < filteredPoints.length; i++) {
+                let dataPoint = filteredPoints[i];
+                let nextDataPoint = filteredPoints[i + 50] ? filteredPoints[i + 50] : filteredPoints[i];
+                if (dataPoint.latitude && dataPoint.longitude && nextDataPoint.latitude && nextDataPoint.longitude) {
+                    const p = fromLonLat([dataPoint.longitude, dataPoint.latitude]);
+                    points.push(p);
+
+
+                    if (dataPoint.flag) {
+                        if (dataPoint.d1) {
+                            trackPointsD1.push(
+                                new Feature({
+                                    distance: dataPoint.d1,
+                                    geometry: new Point(p)
+                                })
+                            )
+                        }
+
+                        if (dataPoint.d2) {
+                            trackPointsD2.push(
+                                new Feature({
+                                    distance: dataPoint.d2,
+                                    geometry: new Point(p)
+                                })
+                            );
+                        }
+                    } else {
+                        if (dataPoint.d1) {
+                            trackPointsUntaggedD1.push(
+                                new Feature({
+                                    distance: dataPoint.d1,
+                                    geometry: new Point(p)
+                                })
+                            )
+                        }
+
+                        if (dataPoint.d2) {
+                            trackPointsUntaggedD2.push(
+                                new Feature({
+                                    distance: dataPoint.d2,
+                                    geometry: new Point(p)
+                                })
+                            );
+                        }
+                    }
+                }
+            }
+
+            //Simplify to 1 point per 2 meter
+            let lineString = new LineString(points);
```

Comment:
We don't need to store `lineString` when we only use it once in the next line anyway